packages/compartment-mapper/src/node-modules.js

```diff
@@ -228,6 +285,8 @@ const inferParsers = (descriptor, location) => {
  * @param {Object} packageDetails.packageDescriptor
  * @param {Set<string>} tags
  * @param {boolean} dev
+ * @param {Array<string>} logicalPath
+ * @param {Map<string, Array<string>>} preferredPackageLogicalPathMap
  * @returns {Promise<undefined>}
  */
 const graphPackage = async (
```

Comment:
Changes to graphPackage are temporary.   
I've ported the algorithm for identifying dependencies from lavamoat/aa for comparison.